hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/init/ExecutionPlan.java

```diff
@@ -378,23 +379,160 @@ private void initDag(InternalSerializationService jobSerializationService) {
         return processors;
     }
 
+    private List<OutboundEdgeStream> createOutboundEdgeStreams(
+            VertexDef vertex,
+            int processorIdx,
+            InternalSerializationService jobSerializationService
+    ) {
+        List<OutboundEdgeStream> outboundStreams = new ArrayList<>();
+        for (EdgeDef edge : vertex.outboundEdges()) {
+            OutboundCollector outboundCollector = createOutboundCollector(edge, processorIdx, jobSerializationService);
+            OutboundEdgeStream outboundEdgeStream = new OutboundEdgeStream(edge.sourceOrdinal(), outboundCollector);
+            outboundStreams.add(outboundEdgeStream);
+        }
+        return outboundStreams;
+    }
+
     /**
-     * Populates {@code localConveyorMap}, {@code edgeSenderConveyorMap}.
-     * Populates {@link #senderMap} and {@link #tasklets} fields.
+     * Each edge is represented by an array of conveyors between the producers and consumers.
+     * There are as many conveyors as there are consumers.
+     * Each conveyor has one queue per producer.
+     *
+     * For a distributed edge, there is one additional producer per member represented
+     * by the ReceiverTasklet.
      */
-    private List<OutboundEdgeStream> createOutboundEdgeStreams(VertexDef srcVertex, int processorIdx,
-                                                               InternalSerializationService jobSerializationService) {
-        final List<OutboundEdgeStream> outboundStreams = new ArrayList<>();
-        for (EdgeDef edge : srcVertex.outboundEdges()) {
-            Map<Address, ConcurrentConveyor<Object>> memberToSenderConveyorMap = null;
-            if (edge.getDistributedTo() != null) {
-                memberToSenderConveyorMap =
-                        memberToSenderConveyorMap(edgeSenderConveyorMap, edge, jobSerializationService);
+    private OutboundCollector createOutboundCollector(
+            EdgeDef edge,
+            int processorIndex,
+            InternalSerializationService jobSerializationService
+    ) {
+        if (edge.isDistributed() && edge.routingPolicy() == RoutingPolicy.ISOLATED) {
+            throw new IllegalArgumentException("Isolated edges must be local: " + edge);
+        }
+        if ((!edge.isDistributed() || !edge.getDistributedTo().equals(DISTRIBUTE_TO_ALL))
```

Comment:
If the second condition is true, the first is always true:
```suggestion
        if (!DISTRIBUTE_TO_ALL.equals(edge.getDistributedTo())
```